src/common/state/error_table.h

```diff
@@ -18,14 +18,19 @@ typedef enum {
   /** An object was added with a different hash from the existing
    *  one. */
   OBJECT_HASH_MISMATCH_ERROR_INDEX = 0,
+  /** An object that was created through a ray.put is lost. */
+  PUT_RECONSTRUCTION_ERROR_INDEX,
   /** The total number of error types. */
   MAX_ERROR_INDEX
 } error_index;
 
 /** Information about the error to be displayed to the user. */
-static const char *error_types[] = {"object_hash_mismatch"};
+static const char *error_types[] = {"object_hash_mismatch",
+                                    "put_reconstruction"};
 static const char *error_messages[] = {
-    "A nondeterministic task was reexecuted."};
+    "A nondeterministic task was reexecuted.",
+    "A task created an object using `ray.put` that was evicted and could not "
+    "be reconstructed. The driver may need to be restarted."};
```

Comment:
Sure. The "data" field in the error message is going to be the the function name though.